src/kernel/src/fs/dev/cdev.rs

```diff
@@ -68,8 +68,8 @@ impl CharacterDevice {
         }
     }
 
-    pub fn sw(&self) -> &CdevSw {
-        self.sw.as_ref()
+    pub fn driver(&self) -> &dyn Device {
+        self.driver.as_ref()
```

Comment:
The `driver` should not exposed to the outside otherwise its methods can be called with incorrect `CharacterDevice`.